payment_paybox/models/account_invoice.py

```diff
@@ -0,0 +1,127 @@
+# coding: utf-8
+from openerp.osv import osv, fields
+from datetime import datetime
+import logging
+import psycopg2
+
+logger = logging.getLogger(__name__)
+
+
+class Invoice(osv.Model):
+
+    _inherit = 'account.invoice'
+
+    # Just add these two fields to display them on portal invoice
+    _columns = {'paid_date': fields.date("Payment date"),
+                'paid_amount': fields.float("Amount paid")}
+
+    def get_invoice_id(self, cr, uid, ref, montant, context=None):
+        """ search and return invoice id for the given reference """
+        invoice_ids = self.search(cr, uid, [('number', '=', ref)])
+        if not invoice_ids:
+            warning_id = self.pool.get('paybox.warning').create(
+                cr, uid, {'ref': ref, 'amount': montant})
+            self.pool.get('paybox.warning').send_warning_mail(cr, uid, warning_id, 'invoice')
+            return False
+        return invoice_ids[0]
+
+    def _portal_payment_block(self, cr, uid, ids, fieldname, arg, context=None):
+        """ invoice residual amount is used instead of total amount """
+        result = dict.fromkeys(ids, False)
+        payment_acquirer = self.pool.get('payment.acquirer')
+        for this in self.browse(cr, uid, ids, context=context):
+            if(this.type == 'out_invoice' and this.state not in ('draft', 'done')
+               and not this.reconciled):
+                result[this.id] = payment_acquirer.render_payment_block(
+                    cr, uid, this, this.number, this.currency_id, this.residual, context=context)
+        return result
+
+    def create_move(self, cr, uid, invoice):
+        """ create move with the right values """
+        journal = self.pool.get('account.journal')
+        period = self.pool.get('account.period')
+        move = self.pool.get('account.move')
+        today = datetime.today()
+        bank_journal_id = journal.search(cr, uid, [('code', '=', 'BNK2')])
+        period_id = invoice.period_id.id if invoice else period.find(cr, uid, today)[0]
+        values = {'journal_id': bank_journal_id[0], 'period_id': period_id,
+                  'date': today, 'name': '/'}
+        move_id = move.create(cr, uid, values)
+        return move_id
+
+    def create_move_lines(self, cr, uid, invoice, move_id, montant):
+        """ create move lines for the given amount and linked to given move_id """
```

Comment:
Why are you creating these move lines? payment modules aren't supposed to create it, but only handle the communication itself with the payment acquirer. Maybe you should create a bridge module (that depends on l10n_fr) that facilitates the accounting with this module in France, but in a split way